pkg/controller/logstash/sset/list.go

```diff
@@ -0,0 +1,116 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package sset
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/logstash/labels"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+	ulog "github.com/elastic/cloud-on-k8s/v2/pkg/utils/log"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/set"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/stringsutil"
+)
+
+type StatefulSetList []appsv1.StatefulSet
+
+// RetrieveActualStatefulSets returns the list of existing StatefulSets labeled for the given es cluster.
+// It is sorted using a natural order sort so that algorithms which are using the resulting list are more predictable and stable.
+func RetrieveActualStatefulSets(c k8s.Client, ls types.NamespacedName) (StatefulSetList, error) {
```

Comment:
This file looks similar to elasticsearch/sset/list.go except the labels. Can these functions move to common?